java/src/Ice/src/main/java/com/zeroc/Ice/InvocationFuture.java

```diff
@@ -35,25 +56,39 @@ public abstract class InvocationFuture<T> extends CompletableFuture<T> {
    *
    * @return The connection.
    */
-  public abstract Connection getConnection();
+  public Connection getConnection() {
+    return null;
+  }
 
   /**
    * Returns the proxy that was used to start the asynchronous invocation, or nil if this object was
    * not obtained via an asynchronous proxy invocation.
    *
    * @return The proxy.
    */
-  public abstract ObjectPrx getProxy();
+  public ObjectPrx getProxy() {
+    return null;
+  }
 
   /**
    * Returns the name of the operation.
    *
    * @return The operation name.
    */
-  public abstract String getOperation();
+  public final String getOperation() {
+    return _operation;
+  }
 
   /** Blocks the caller until the result of the invocation is available. */
-  public abstract void waitForCompleted();
+  public final void waitForCompleted() {
```

Comment:
It's not clear to me why more (most) methods are not final. For example, should getCommunicator be final?

Reply:
It's quite likely they could be. And probably many other places as well...
I'll look into it in a later PR!